### PR TITLE
[Context] Allow flexible Contexts to be utilized

### PR DESCRIPTION
**Pull request by ShootingKing-AM**
Allow Childs of Context class to be used in cmdClient. 
This lets users of this lib to hook/override certain functions of Contexts, rendering the library more flexible.

### CODE REVIEW

**Comment by Intery** [comment]
Thanks for this! The `baseContext` should always have been stored that way, but no-one picked up on it before.

By the way `Optional[X]` in python typing is the same thing as `Union[X, None]`, so `baseContext` actually *shouldn't* be optional, with `Context` as the default value. The comment `type: Type[Context]` is also sufficient for type-ing constructor arguments (see PEP 484) although it doesn't hurt to have it explicitly annotated as well.

File: cmdClient.py [hunk 1]
```diff
@@ -26,13 +26,13 @@ class cmdClient(discord.Client):
 
     cmd_names = {}  # Command name cache, {cmdname: Command}, including aliases.
 
-    def __init__(self, prefix=None, owners=None, ctx_cache=None, baseContext=Context, **kwargs):
+    def __init__(self, prefix=None, owners=None, ctx_cache=None, baseContext: Optional[Type[Context]] = Context, **kwargs):
```
Comment (Intery) [inline cmdClient.py]:
```suggestion
    def __init__(self, prefix=None, owners=None, ctx_cache=None, baseContext: Type[Context] = Context, **kwargs):
```

**Comment by ShootingKing-AM** [comment]
> Thanks for this! The `baseContext` should always have been stored that way, but no-one picked up on it before.
> 
> By the way `Optional[X]` in python typing is the same thing as `Union[X, None]`, so `baseContext` actually _shouldn't_ be optional, with `Context` as the default value. The comment `type: Type[Context]` is also sufficient for type-ing constructor arguments (see PEP 484) although it doesn't hurt to have it explicitly annotated as well.

I mean, even if they dont pass any arg for Context, you can cut them some slack for them by init'ing with default Context. 
Personally, I am OK with either :D

**Comment by Intery** [comment]
Oh definitely! It's a keyword argument that defaults to `Context` though, so it's automatically an optional argument (perhaps "optional type" is a bit confusing there, but that's what they chose for values that can be `None`).

Thanks for the PR!

**Comment by ShootingKing-AM** [comment]
Oh, i was confused with the Optional thingy xD

Please do update the StudyLion/master and staging with this change, cause I actually need this flexibility for topgg-feature.

**Pull request merged**